src/periodic.jl

```diff
@@ -0,0 +1,95 @@
+"""
+    Periodic{T}(f, period::T, phase_shift::T) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+# Fields
+* `f::Union{typeof(cos), typeof(sin)}`: the periodic function
+* `period<:Union{Real, Period}`: the function period. Must be strictly positive.
+* `phase_shift<:Union{Real, Period}`: adjusts the phase of the periodic function, measured
+    in the same units as the input. Increasing the value translates the function to the
+    right, toward higher/later input values.
+"""
+struct Periodic{T} <: Transform where T <: Union{Real, Period}
+    f::Union{typeof(cos), typeof(sin)}
+    period::T
+    phase_shift::T
+
+    function Periodic(f, period, phase_shift)
+        if period <= zero(typeof(period))
```

Comment:
```suggestion
        period <= zero(T) && throw(ArgumentError("period must be strictly positive."))
```

might also fit on one line